test/ts/integration/token_registry.ts

```diff
@@ -0,0 +1,179 @@
+// External Libraries
+import * as chai from "chai";
+import { BigNumber } from "bignumber.js";
+
+// Test Utils
+import { BigNumberSetup } from "../test_utils/bignumber_setup";
+import ChaiSetup from "../test_utils/chai_setup";
+import { multiSigExecute } from "../test_utils/utils";
+
+// Wrappers
+import { TokenRegistryContract } from "../../../types/generated/token_registry";
+import { MultiSigWalletContract } from "../../../types/generated/multi_sig_wallet";
+import { Address, TxData } from "../../../types/common";
+import { BaseContract } from "../../../types/base_contract";
+
+ChaiSetup.configure();
+
+// Configure BigNumber exponentiation
+BigNumberSetup.configure();
+
+const expect = chai.expect;
+
+contract("Token Registry (Integration Tests)", async (ACCOUNTS) => {
+    const CONTRACT_OWNER = ACCOUNTS[0];
+    const TX_DEFAULTS = { from: CONTRACT_OWNER, gas: 4000000 };
+
+    let registry: TokenRegistryContract;
+
+    before(async () => {
+        registry = await TokenRegistryContract.deployed(web3, TX_DEFAULTS);
+    });
+
+    describe("#getTokenNameBySymbol", () => {
+        describe("when given 'WETH'", () => {
+            it("returns 'Canonical Wrapped Ether", async () => {
+                expect(await registry.getTokenNameBySymbol.callAsync("WETH")).to.equal(
+                    "Canonical Wrapped Ether",
+                );
+            });
+        });
+
+        describe("when given 'REP'", () => {
+            it("returns 'Augur REP", async () => {
+                expect(await registry.getTokenNameBySymbol.callAsync("REP")).to.equal("Augur REP");
+            });
+        });
+    });
+
+    describe("#getNumDecimalsFromSymbol", () => {
+        describe("when given 'WETH'", () => {
+            it("returns '18'", async () => {
+                const numDecimals = await registry.getNumDecimalsFromSymbol.callAsync("WETH");
+
+                expect(numDecimals.toNumber()).to.equal(18);
+            });
+        });
+
+        describe("when given 'TRX'", () => {
+            it("returns '6", async () => {
+                const numDecimals = await registry.getNumDecimalsFromSymbol.callAsync("TRX");
+
+                expect(numDecimals.toNumber()).to.equal(6);
+            });
+        });
+    });
+
+    describe("#getTokenIndexBySymbol", () => {
+        describe("when given 'WETH'", () => {
+            it("returns a number", async () => {
+                const index = await registry.getTokenIndexBySymbol.callAsync("WETH");
+
+                expect(index.toNumber()).to.be.at.least(0);
+            });
+        });
+    });
+
+    describe("#getTokenAddressBySymbol", () => {
+        describe("when given 'WETH'", () => {
+            it("returns a valid address", async () => {
+                const address = await registry.getTokenAddressBySymbol.callAsync("WETH");
+
+                expect(address.length).to.eq(42);
```

Comment:
This should test for something other than length -- this test will pass silently even if `address(0)` is returned, which is what would happen when an untracked symbol is queried.

Reply:
e.g. `.not.eq(NULL_ADDRESS)`